File: BCKEND/Java/SistemaBancario_SPRING/src/main/java/edu/senac/pe/SistemaBancario_SPRING/model/Cliente.java
package edu.senac.pe.SistemaBancario_SPRING.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
public class Cliente {
	@Id
    @Column(nullable = false, length = 14, name = "cpf")
    private String cpf;

    @Column(nullable = false, length = 100, name = "nome")
    private String nome;

    @Column(nullable = false, columnDefinition = "date", name = "dataNasc")
    private String dataNasc;

    @Column(nullable = false, columnDefinition = "int", name = "rg")
    private Integer rg;

    @Column(nullable = false, length = 45, unique = true, name = "email")
    private String email;

    @Column(nullable = false, length = 25, unique = true, name = "telefone")
    private String telefone;

    // Construtor padrão exigido pelo JPA
    public Cliente() {
    }

    // Construtor completo (mantém o seu original)
    public Cliente(String nome, String cpf, String dataNasc, Integer rg, String email, String telefone) {
        this.nome = nome;
        this.cpf = cpf;
        this.dataNasc = dataNasc;
        this.rg = rg;
        this.email = email;
        this.telefone = telefone;
    }
	
	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getTelefone() {
		return telefone;
	}

	public void setTelefone(String telefone) {
		this.telefone = telefone;
	}

	/*public Endereco getEndereco() {
		return endereco;
	}

	public void setEndereco(Endereco endereco) {
		this.endereco = endereco;
	}*/

	public String getCpf() {
		return cpf;
	}

	public String getDataNasc() {
		return dataNasc;
	}

	public Integer getRg() {
		return rg;
	}

	@Override
	public String toString() {
		return "Cliente, " + nome + ", CPF: " + cpf + ", nacido em " + dataNasc + ", portador do RG: " + rg + ", e-mail: " + email + ", telefone: "
				+ telefone; // + ", endereco: " + endereco;
	}
}

File: BCKEND/TypeScript/BancoFinanceiroNodeExpress/src/models/Cliente.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';

@Entity('clientes')
export class Cliente {
    @PrimaryColumn({ length: 14 })
    cpf: string;

    @Column({ length: 150 })
    nome: string;

    @Column({ type: 'date' })
    dataNascimento: string;

    @Column({ length: 20 })
    rg: string;

    @Column({ length: 150, unique: true })
    email: string;

    @Column({ length: 20, unique: true })
    telefone: string;

    constructor(nome: string, cpf: string, dataNascimento: string, rg: string, email: string, telefone: string) {
        this.nome = nome;
        this.cpf = cpf;
        this.dataNascimento = dataNascimento;
        this.rg = rg;
        this.email = email;
        this.telefone = telefone;
    }

    public getNome(): string {
        return this.nome;
    }

    public getCpf(): string {
        return this.cpf;
    }

    public getDataNascimento(): string {
        return this.dataNascimento;
    }

    public getRg(): string {
        return this.rg;
    }

    public getEmail(): string {
        return this.email;
    }

    public getTelefone(): string {
        return this.telefone;
    }

    public setNome(nome: string): void {
        this.nome = nome;
    }

    public setEmail(email: string): void {
        this.email = email;
    }

    public setTelefone(telefone: string): void {
        this.telefone = telefone;
    }

    public toString(): string {
        return `Nome: ${this.nome}, CPF: ${this.cpf}, Data de Nascimento: ${this.dataNascimento}, RG: ${this.rg}, Email: ${this.email}, Telefone: ${this.telefone}]`;
    }
}